### PR TITLE
Add image build to regression

### PR DESCRIPTION
**Pull request by mdhorn**
Changes proposed in this pull request:
- Add building a baseline image file as a regression

### CODE REVIEW

**Comment by mdhorn** [comment]
@otaviobp This is the test swupd-client should run before merge/releaase.

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,10 @@ endif
 export GOPATH=$(pkg_dir)
 export GO_PACKAGE_PREFIX := github.com/clearlinux/clr-installer
 export TESTS_DIR := $(top_srcdir)/tests/
+export BASEBUILD_DIR := $(shell mktemp --dir --dry-run)
+export BASEIMGNAME := baseline
+export BASEIMG := $(top_srcdir)/tests/$(BASEIMGNAME)
```
Comment (jwakre) [inline Makefile]: nit: Replace `$(top_srcdir)/tests` with `$(TESTS_DIR)`

Otherwise +1

File: Makefile [hunk 2]
```diff
@@ -190,6 +194,30 @@ check: gopath bundle-check
 		/bin/false ; \
 	fi; \
 
+check-image: gopath build-tui
+	@if [ "${runuid}" != "0" ] ; then \
+		echo "check-image needs to be run as root; please use 'sudo'"; \
+		/bin/false; \
+	fi
+	@mkdir -p ${BASEBUILD_DIR}
+	cd ${BASEBUILD_DIR} ; ${LOCAL_GOPATH}/bin/clr-installer-tui --config ${TESTS_DIR}/${BASEIMGNAME}.yaml --log-file ${BASEIMGLOG}
```
Comment (jwakre) [inline Makefile]: s/`${TESTS_DIR}/${BASEIMGNAME}.yaml` /`$(BASEIMG).yaml`

**Comment by mdhorn** [comment]
I'm going to wait to merge until I know @otaviobp can use this as it was targeted for swupd usae.

**Pull request merged**